### PR TITLE
ODC-6443: Add adv. option to customize route labels

### PR DESCRIPTION
**Pull request by jerolimov**
**Fixes**: 
https://issues.redhat.com/browse/ODC-6443

**Analysis / Root cause**: 
As a user, I want have the option to add additional labels to a Route, as I could do in OCP3. See RFE-622.

**Solution Description**: 
Added a new "Labels" field to advanced route option in the add/import flow and the edit page as well.

It is NOT shown for Serverless routes because these are auto-generated and don't accept custom labels. At least exclusive labels for the route.

**Screen shots / Gifs for design review**: 
@openshift/team-devconsole-ux
@serenamarie125 

Default routing options:

image

When opening the advanced routing options:

image

**Unit test coverage report**: 
- [x] Added some unit and e2e tests.

**Test setup:**
1. Import from git or container image
    - Open the "advanced routing options"
    - Enter a label
2. Open the Deployment and route detail pages
    - The Deployment must not contain the route-specific labels
    - The Route should contain them :smirk: 
3. Select "Edit your-deployment" and update the labels should also update them.

**Browser conformance**: 
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
@jerolimov frontend is failing. You need to update the i18n files.

**Comment by jerolimov** [comment]
> @jerolimov frontend is failing. You need to update the i18n files.

Thanks for that hint. Updated just the i18n file.

**Comment by jerolimov** [comment]
> ✖  crud/other-routes.spec.ts
> AssertionError: 1 accessibility violation was detected

Will check that.

**Comment by jerolimov** [comment]
Could not reproduce this locally.

image

/retest

**Comment by jerolimov** [comment]
Run the cypress tests once locally and esp. the new tests passes:

Screenshot from 2021-12-16 14-10-37

Screenshot from 2021-12-16 14-12-02

**Comment by divyanshiGupta** [comment]
@jerolimov I have tested the PR and everything seems to work fine. But the e2e tests are failing for some reason. Lets see if it passes this time otherwise we will have to look into why its failing.

**Comment by sanketpathak** [comment]
Tests are passing on my local, Ran tests related to the epic ODC-6266
Screenshot from 2021-12-16 20-08-08

File: frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts [hunk 1]
```diff
@@ -175,6 +175,31 @@ export const getKsvcRouteData = (resource: K8sResourceKind) => {
...
+
+export const getRouteLabels = (
+  route: K8sResourceKind,
+  resource: K8sResourceKind,
+): Record<string, string> => {
+  const allLabels = _.get(resource, 'metadata.labels', {});
+  const allRouteLabels = _.get(route, 'metadata.labels', {});
+  const filteredRouteLabels = _.omit(allRouteLabels, [
+    ...getDefaultLabels(),
+    ...Object.keys(allLabels),
```
Comment (divyanshiGupta) [inline frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts:198]: @jerolimov `allLabels` will have the `defaultLabels` as well so there is no need for adding `defaultLabels` twice.

**Review by divyanshiGupta — CHANGES_REQUESTED** [review]

**Comment by karthikjeeyar** [comment]
Verified locally, works as expected.

File: frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts (hunk 1 above)
Comment (jerolimov) [reply frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts:198]: @divyanshiGupta Technical you are right here, but I think it's good to remove everything from our sourcecode driven list as well. For example, if we change or add another label here. If you not 100% against this I would like to keep it. Wdyt? :)

**Comment by jerolimov** [comment]
Propagate from epic:
/label docs-approved
/label px-approved

**Pull request merged**